packages/core/http/core-http-router-server-internal/src/response.ts

```diff
@@ -24,10 +24,6 @@ import type {
 } from '@kbn/core-http-server';
 import mime from 'mime';
 
-export function isKibanaResponse(response: Record<string, any>): response is IKibanaResponse {
-  return typeof response.status === 'number' && typeof response.options === 'object';
```

Comment:
Not the most thorough check and would be nice to add unit tests for before making public but overall happy with the idea.

Reply:
@jloleysens I added some tests in https://github.com/elastic/kibana/pull/182392/commits/d1a8d60ab5f1750278221d23155c8b0dae74b554 and made some small changes to the helper to make it a bit more strict, let me know what you think!